app_dart/lib/src/service/scheduler.dart

```diff
@@ -256,3 +257,57 @@ class Scheduler {
     }
   }
 }
+
+/// Load [yamlConfig] to [SchedulerConfig] and validate the dependency graph.
+SchedulerConfig loadSchedulerConfig(YamlMap yamlConfig) {
+  final SchedulerConfig config = SchedulerConfig();
+  config.mergeFromProto3Json(yamlConfig);
+  _validateSchedulerConfig(config);
+
+  return config;
+}
+
+void _validateSchedulerConfig(SchedulerConfig schedulerConfig) {
+  if (schedulerConfig.targets.isEmpty) {
+    throw const FormatException('Scheduler config must have at least 1 target');
+  }
+
+  if (schedulerConfig.enabledBranches.isEmpty) {
+    throw const FormatException('Scheduler config must have at least 1 enabled branch');
+  }
+
+  final Map<String, List<Target>> targetGraph = <String, List<Target>>{};
+  final List<String> exceptions = <String>[];
+  // Construct [targetGraph]. With a one scan approach, cycles in the graph
+  // cannot exist as it only works forward.
+  for (final Target target in schedulerConfig.targets) {
+    if (targetGraph.containsKey(target.name)) {
+      exceptions.add('ERROR: ${target.name} already exists in graph');
+    } else {
+      targetGraph[target.name] = <Target>[];
+      // Add edges
+      if (target.dependencies.isNotEmpty) {
+        if (target.dependencies.length != 1) {
```

Comment:
why is `dependencies` defined as repeated then?

Reply:
This is future functionality we'd like to support, but it adds a lot of complexity to the initial scheduler. It's easier to support if if this is already a list